Source/WarpX.cpp

```diff
@@ -141,8 +141,8 @@ int WarpX::current_centering_noz = 2;
 bool WarpX::use_fdtd_nci_corr = false;
 bool WarpX::galerkin_interpolation = true;
 
-bool WarpX::use_filter        = false;
-bool WarpX::use_kspace_filter       = false;
+bool WarpX::use_filter = true;
+bool WarpX::use_kspace_filter = true;
```

Comment:
Should this switch be generally `false` actually unless explicitly set?
```suggestion
bool WarpX::use_kspace_filter = false;
```

Reply:
Yes, that is what I had initially and I think it is safer to leave it false, maybe adding a comment that points to the code below where it is set equal to `use_filter`. So that we avoid exactly the issue observed here.

Reply:
The problem is probably being caused in `Source/Diagnostics/ComputeDiagFunctors/RhoFunctor.cpp` which uses `use_kspace_filter`. The `if` statement should also check if `WarpX::maxwell_solver_id == MaxwellSolverAlgo::PSATD`. I still think that `use_kspace_filter` should be set to `true` since that is its true default value.

Reply:
OK, I will try to fix the issue in the rho functor, and see if that works fine then. Thanks for the feedback!

Reply:
@dpgrote Thank you, Dave, for pointing to the missing runtime check. Done in https://github.com/ECP-WarpX/WarpX/pull/2031/commits/176b860402cced823d780c8a7862eb2ebbc751a5. 
@ax3l We reverted again, because Dave identified the innermost issue in the code. It should be OK now. Let's see if CI passes.